enterprise/backend/src/metabase_enterprise/semantic_search/index.clj

```diff
@@ -226,29 +246,85 @@
     (when (seq conditions)
       (into [:and] conditions))))
 
+(defn- keyword-search-query [index search-context]
+  (let [filters (search-filters search-context)
+        ts-search-expr (search/to-tsquery-expr (:search-string search-context))
+        tsv-lang (search/tsv-language)]
+    {:select [[:id :id]
+              [:model_id :model_id]
+              [:model :model]
+              [:content :content]
+              [:verified :verified]
+              [:metadata :metadata]
+              [[:raw "row_number() OVER (ORDER BY ts_rank_cd(text_search_vector, query) DESC)"] :keyword_rank]]
+     :from [(keyword (:table-name index))]
+     ;; Using a join allows us to share the query expression between our SELECT and WHERE clauses.
+     ;; This follows the same secure pattern as metabase.search.appdb.specialization.postgres/base-query
+     :join [[[:raw "to_tsquery('" tsv-lang "', " [:lift ts-search-expr] ")"]
```

Comment:
gut check: it looks like `search/to-tsquery-expr` is doing some escaping on the input (and this expression basically matches the one in appdb backend)?

Reply:
yeah, this should be essentially the same as what the appdb backend is doing